Modules/timemodule.c

```diff
@@ -2044,6 +2048,23 @@ PyInit_time(void)
     return PyModuleDef_Init(&timemodule);
 }
 
+#ifndef MS_WINDOWS
+static int
+sleep_unix(struct timeval timeout)
+{
+    int ret = 0;
+#ifdef HAVE_CLOCK_NANOSLEEP
+    struct timespec request;
+    request.tv_sec = timeout.tv_sec;
+    request.tv_nsec = (long)(timeout.tv_usec) * 1000;
```

Comment:
This line makes me sad. The caller uses nanosecond resolution (10^-9), then _PyTime_AsTimeval() truncates to microsecond resolution (10^-6).

I would prefer to inline this code in pysleep() and avoid _PyTime_AsTimeval() for the clock_nanosleep() code path.